starfish/intensity_table/intensity_table.py

```diff
@@ -419,49 +430,46 @@ def to_decoded_spots(self) -> DecodedSpots:
         return DecodedSpots(df)
 
-    def to_expression_matrix(self, regions: Optional[regional.many]=None) -> ExpressionMatrix:
+    def to_expression_matrix(self) -> ExpressionMatrix:
         """Generates a cell x gene count matrix where each cell is annotated with spatial metadata
 
-        Parameters
-        ----------
-        regions: Optional[regional.Many]
-            cell segmentation results that were used to assign points to cells. If not provided, the
-            centers of the cells will be estimated by taking the midpoint between the extreme-valued
-            spots on each axis.
+        Requires that spots in the IntensityTable have been assigned to cells.
 
         Returns
         -------
         ExpressionMatrix :
             cell x gene expression table
         """
-
-        # create the 2-d counts matrix
-        grouped = self.to_features_dataframe().groupby(['cell_id', 'target'])
+        try:
+            grouped = self.to_features_dataframe().groupby([Features.CELL_ID, Features.TARGET])
+        except KeyError as e:
+            if Features.CELL_ID in str(e):
+                raise RuntimeError(
+                    "IntensityTable must have 'cell_id' assignments for each cell before "
+                    "this function can be called. See starfish.TargetAssignment.Label."
+                )
+            else:
+                raise
         counts = grouped.count().iloc[:, 0].unstack().fillna(0)
-
-        if regions:
-            # counts.index stores cell_id, extract cell information from the regional.many object
-            metadata = {
-                "area": ("cells", [regions[id_].area for id_ in counts.index]),
-                "x": ("cells", [regions[id_].center[0] for id_ in counts.index]),
-                "y": ("cells", [regions[id_].center[1] for id_ in counts.index]),
-                "z": ("cells", np.zeros(counts.shape[0]))
-            }
+        if self.has_physical_coords:
+            grouped = self.to_features_dataframe().groupby([Features.CELL_ID])[[
+                Axes.X, Axes.Y, Axes.ZPLANE, Coordinates.X, Coordinates.Y, Coordinates.Z]]
         else:
-            grouped = self.to_features_dataframe().groupby(['cell_id'])[['x', 'y', 'z']]
-            min_ = grouped.min()
-            max_ = grouped.max()
-            coordinate_df = min_ + (max_ - min_) / 2
-            metadata = {name: ("cells", data.values) for name, data in coordinate_df.items()}
-            metadata['area'] = ("cells", np.full(counts.shape[0], fill_value=np.nan))
+            grouped = self.to_features_dataframe().groupby([Features.CELL_ID])[[
+                Axes.X, Axes.Y, Axes.ZPLANE]]
+        min_ = grouped.min()
+        max_ = grouped.max()
+        coordinate_df = min_ + (max_ - min_) / 2
+        metadata = {name: (Features.CELLS, data.values) for name, data in coordinate_df.items()}
+        metadata[Features.AREA] = (Features.CELLS, np.full(counts.shape[0], fill_value=np.nan))
```

Comment:
I know you're propagating this, but it seems like it needs a `#TODO fixme` ;-)

@kne42 is this `AREA = np.nan` getting fixed in your work on segmentation?

Reply:
no – will this be a problem?